src/nvme/tree.c

```diff
@@ -1864,19 +1864,45 @@ static nvme_ns_t nvme_ns_open(const char *name)
 	return NULL;
 }
 
+static inline bool nvme_ns_is_generic(const char *name)
+{
+	int instance, head_instance;
+
+	if (sscanf(name, "ng%dn%d", &instance, &head_instance) != 2)
+		return false;
+	return true;
+}
+
+static const char *nvme_ns_generic_to_blkdev(const char *generic)
+{
+
+	int instance, head_instance;
+	char blkdev[PATH_MAX];
+
+	if (!nvme_ns_is_generic(generic))
+		return generic;
+
+	sscanf(generic, "ng%dn%d", &instance, &head_instance);
+	sprintf(blkdev, "nvme%dn%d", instance, head_instance);
+
+	return strdup(blkdev);
+}
+
 static struct nvme_ns *__nvme_scan_namespace(const char *sysfs_dir, const char *name)
 {
 	struct nvme_ns *n;
 	char *path;
 	int ret;
+	char *blkdev;
 
-	ret = asprintf(&path, "%s/%s", sysfs_dir, name);
+	blkdev = (char *)nvme_ns_generic_to_blkdev(name);
```

Comment:
`blkdev` is leaked here if it's it was a generic device name. I would just update the function signature to return 'char *' and do a `strdup` on generic as well and also  free `blkdev` after the `asprintf` call.

Reply:
Sure, let me update. Thanks!

Reply:
Updated.